src/coreclr/inc/corinfo.h

```diff
@@ -2132,6 +2132,21 @@ class ICorStaticInfo
         CORINFO_METHOD_HANDLE meth1Hnd,
         CORINFO_METHOD_HANDLE meth2Hnd) = 0;
 
+    //------------------------------------------------------------------------------
+    // getTypeDefinition: Get the (unconstructed) type definition from a given type handle.
+    //
+    // Arguments:
+    //    type - The input type handle
+    //
+    // Return Value:
+    //   The type handle for the (unconstructed) type definition from type.
+    //
+    // Remarks:
+    //   This is equivalent of Type.GetGenericTypeDefinition(). Given a generic type handle, it will
+    //   return the original type definition (eg. for Foo<int> it will return Foo<>). If the input
+    //   type handle represents a type that is not generic, it will just return the same type.
```

Comment:
Note that this does not match behavior of `RuntimeType.GetGenericTypeDefinition`. `RuntimeType.GetGenericTypeDefinition()` throws for non-generic types.

Reply:
I've added a check in 839f4c39ad30f1a8cc74b72243360624565588d6 to avoid expanding the intrinsic if the type argument in the `typeof` expression is not a generic type, and updated the remarks in this method to clarify that it should only be called when the input type is generic. Is that ok?

Reply:
Test case that verifies that it continues throwing would be nice. Looks good otherwise.